localflavor/nl/models.py

```diff
@@ -1,62 +1,88 @@
-        'invalid': _('Enter a valid bank account number'),
-        'wrong_length': _('Bank account numbers have 1 - 7, 9 or 10 digits'),
-    }
+    def __init__(self, *args, **kwargs):
+        kwargs['max_length'] = 7
+        super(NLZipCodeField, self).__init__(*args, **kwargs)
 
-    def __init__(self, regex=None, message=None, code=None):
-        super(NLBankAccountNumberFieldValidator, self).__init__(regex='^[0-9]+$',
-                                                                message=self.default_error_messages['invalid'])
-        self.no_leading_zeros_regex = re.compile('[1-9]+')
+    def to_python(self, value):
+        return '%s %s' % (value[:4], value[4:])
+
+    def formfield(self, **kwargs):
+        defaults = {'form_class': forms.NLZipCodeField}
+        defaults.update(kwargs)
+        return super(NLZipCodeField, self).formfield(**defaults)
+
+
+class NLProvinceField(models.CharField):
+    description = _('Dutch social security number (SoFi)')
+
+    def __init__(self, *args, **kwargs):
+        kwargs['choices'] = PROVINCE_CHOICES
+        super(NLProvinceField, self).__init__(*args, **kwargs)
 
-    def __call__(self, value):
-        super(NLBankAccountNumberFieldValidator, self).__call__(value)
+    def formfield(self, **kwargs):
+        defaults = {'form_class': forms.NLProvinceSelect}
+        defaults.update(kwargs)
+        return super(NLProvinceField, self).formfield(**defaults)
 
-        # Need to check for values over the field's max length before the zero are stripped.
-        # This check is needed to allow this validator to be used without Django's MaxLengthValidator.
-        if len(value) > 10:
-            raise ValidationError(self.default_error_messages['wrong_length'])
 
-        # Strip the leading zeros.
-        m = re.search(self.no_leading_zeros_regex, value)
-        if not m:
-            raise ValidationError(self.default_error_messages['invalid'])
-        value = value[m.start():]
+class NLSoFiNumberField(models.CharField):
 
-        if len(value) != 9 and len(value) != 10 and not 1 <= len(value) <= 7:
-            raise ValidationError(self.default_error_messages['wrong_length'])
+    description = _('Dutch social security number (SoFi)')
 
-        # Perform the eleven test validation on non-PostBank numbers.
-        if len(value) == 9 or len(value) == 10:
-            if len(value) == 9:
-                value = "0" + value
+    validators = [NLSoFiNumberFieldValidator()]
+
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault('max_length', 12)
+        super(NLSoFiNumberField, self).__init__(*args, **kwargs)
+
+    def formfield(self, **kwargs):
+        defaults = {'form_class': forms.NLSoFiNumberField}
+        defaults.update(kwargs)
+        return super(NLSoFiNumberField, self).formfield(**defaults)
+
+
+class NLPhoneNumberField(models.CharField):
+    """
+    Dutch phone number model field
+    """
+    description = _('Dutch phone number')
```

Comment:
This is in the code multiple times, should I remove them all? Are they not useful while generating the docs?